src/apps/btc/btc_sign.c

```diff
@@ -75,6 +87,18 @@ static void* _hash_outputs_ctx = NULL;
 // https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki step 8.
 static uint8_t _hash_outputs[32] = {0};
 
+static void _maybe_pop_empty_screen(void)
+{
+    if (_empty_component != NULL) {
+        if (ui_screen_stack_top() != _empty_component) {
+            Abort("btc_sign: mismatched screen push/pop");
+        }
+        ui_screen_stack_pop_and_clean();
```

Comment:
I wonder why not call the existing `ui_screen_stack_pop` and forget about it? It should be eventually cleaned up in `ui_screen_stack_cleanup`, right. Or is it a special case because of `_reset()`?

Reply:
On the normal execution with the normal mainloop on the device it works like you suggest, but I didn't want to make the app code rely on having a mainloop environment. The issue shows up in the unit tests of signing, where we don't do the `ui_screen_stack_cleanup()` between runs (we could, but I found this was cleaner).

fyi: `ui_screen_stack_cleanup` is a bad workaround to not run into memory bugs if the component cleans itself during execution of a button-press callback. we will hopefully remove it at some point.

In this case, cleaning it immediately is safe.

WDYT?

Reply:
Yeah, makes sense. I like this being explicit. In fact, I found it a bit hard to track where pushed components are cleaned up. I actually didn't find an explicit call to `ui_screen_stack_cleanup` at the end of this signing flow, so I assumed it must be somewhere at a "layer" above it or in a loop where it all starts. 

:+1: